server.js

```diff
@@ -1,2 +1,100 @@
-require('dotenv').config()
-const express = require('express')
\ No newline at end of file
+//server interfaces w front end and database
+require("dotenv").config();
+const express = require("express");
+const path = require("path");
+const app = express();
+const port = process.env.PORT || 5000;
+const staticDir = path.resolve('./client/public');
+const { ObjectId } = require('mongodb');
+
+const mongoose = require("mongoose");
+//connect to the database 
+mongoose.connect("mongodb://localhost:27017/today_i_learned", {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+});
+const todayDB = mongoose.connection;
+//connecting to my today I learned database
+
+//telling the server to show the error if there is one
+todayDB.on("error", console.error.bind(console, "connection error:"));
+
+//template for an entry into my DB
+const entrySchema = new mongoose.Schema({
+  title: String,
+  content: String,
+  author: String,
+  date: Date,
+  tags: String,
+});
+//model
+const PostModel = mongoose.model("posts", entrySchema);
+app.use(express.urlencoded({extended: true}));
+app.use(express.static(staticDir));
+//schema and model have to match
+//endpoint / shows all entries into the journal
+app.post("/add", async (req, res) => {
+  let newEntry = new PostModel({
+    title: req.body.title,
+    content: req.body.content,
+    author: req.body.author,
+    date: req.body.date,
+    tags: req.body.tags,
+  });
+  await newEntry.save();
+  res.status(200).send("New entry added")
```

Comment:
This line seems to be leaving the user on a page that says response submitted and errors out the console. I would comment out line 45 and either change to an alert or remove completely so that the user is returned to the home page after submission